src/modules/AutoInstallRules.rb

```diff
@@ -36,6 +36,14 @@ def main
 
       Yast.include self, "autoinstall/io.rb"
 
+      reset
+      AutoInstallRules()
+    end
+
+    # Reset the module's state
```

Comment:
Do you mean `ProbeRules` method, right? Just because when it's called by first time, it populates `@ATTR`. So when you call it again, it does nothing because the first line of method is `return if @ATTR.size>0`.